src/main/java/decodes/db/DecodesScript.java

```diff
@@ -16,538 +16,721 @@
-			return null;
-		return scriptType.substring(idx+1);
-	}
-	
-	public void addMissing(String symbol)
-	{
-		missingSymbols.add(symbol);
-	}
-	
-	public boolean isMissingSymbol(String symbol)
-	{
-//Logger.instance().debug3("script.isMissingSymbol '" + symbol + "' checking against " + missingSymbols.size()
-//	+ " defined symbols.");
-		return missingSymbols.contains(symbol);
-//		for(String s : missingSymbols)
-//			if (s.equals(symbol))
-//				return true;
-//else Logger.instance().debug3("   doesn't equal '" + s + "'");
-//		return false;
-	}
-
-	public void setIncludePMs(ArrayList<String> includePMs)
-	{
-		this.includePMs = includePMs;
-		if (includePMs != null && includePMs.size() == 0)
-			this.includePMs = null;
-if (this.includePMs != null) Logger.instance().debug1("setIncludePMs: includePMs has " 
-+ includePMs.size() + " strings. [0]=" + includePMs.get(0));
-	}
-	
-	/**
-	 * Add the PMs as sensor values.
-	 */
-	private void addPMs(DecodedMessage decodedMessage)
-	{
-		RawMessage rm = decodedMessage.getRawMessage();
-		if (rm == null)
-			return;
-		Platform p = null;
-		try { p = rm.getPlatform(); }
-		catch(UnknownPlatformException ex) { return; }
-		if (p == null)
-			return;
-		PlatformConfig pc = p.getConfig();
-		if (pc == null)
-			return;
-	  nextPM:
-		for(String pmname : includePMs)
-		{
-Logger.instance().debug1("addPMs looking for '" + pmname + "'");
-			Variable v = rm.getPM(pmname);
-			if (v == null)
-			{
-				Logger.instance().info(
-					"addPMs: Message for '" + new String(rm.getHeader()) 
-					+ "' does not have requested performance measurement '" + pmname + "' -- ignored.");
-				continue;
-			}
-			for (ConfigSensor cs : pc.getSensorVec())
-			{
-				if (cs.sensorName.equalsIgnoreCase(pmname))
-				{
-					decodedMessage.addSample(cs.sensorNumber, v, 0);
-Logger.instance().debug1("addPM: added pm '" + pmname + "' to sensor " + cs.sensorNumber + " with value '" + v.toString() + "'");
-					continue nextPM;
-				}
-			}
-			// Fell through, no matching sensor:
-			Logger.instance().info("addPMs: The platform config for '" + new String(rm.getHeader()) 
-				+ "' does not have a sensor named '" + pmname + "' -- discarded.");
-		}
-	}
-
+    public static final Logger logger = Logger.instance();
+    public static final EmptyDecodesScriptReader EMPTY_SCRIPT = new EmptyDecodesScriptReader();
+    // _id is stored in the IdDatabaseObject superclass.
+
+    /**
+     * Name of this script. Must be unique within a configuration.
+     */
+    public String scriptName;
+
+    /** Script type. The only type thus far implemented is 'DECODES'.*/
+    public String scriptType;
+
+    /** Links */
+    public PlatformConfig platformConfig;
+
+    /**
+    * This stores references to the ScriptSensors.  This data member will
+    * never be null.  Note that the index
+    * of each element of this Vector does not necessarily correspond with
+    * the sensor number.
+    */
+    public Vector<ScriptSensor> scriptSensors;
```

Comment:
Would love to see these typed as `List` while we're making big changes to this class.

Reply:
That's very, that should really break much... in theory.